src/dataframerow/dataframerow.jl

```diff
@@ -80,21 +80,20 @@ index(r::DataFrameRow) = index(parent(r))
 Base.size(r::DataFrameRow) = (size(parent(r), 2),)
 Base.size(r::DataFrameRow, i) = size(r)[i]
 Base.length(r::DataFrameRow) = size(parent(r), 2)
+Base.ndims(r::DataFrameRow) = 1
+Base.ndims(::Type{DataFrameRow{T}}) where {T<:AbstractDataFrame} = 1
 
-Compat.lastindex(r::DataFrameRow) = size(parent(r), 2)
+Base.lastindex(r::DataFrameRow) = size(parent(r), 2)
 
-function Base.iterate(r::DataFrameRow)
-    Base.depwarn("iteration over DataFrameRow will return values in the future:" *
-                 "use pairs(r::DataFrameRow) to get the current behavior",
-                 :start)
-    iterate(r, 1)
-end
+Base.iterate(r::DataFrameRow) = iterate(r, 1)
 
 function Base.iterate(r::DataFrameRow, st)
     st > length(r) && return nothing
     return (r[st], st + 1)
 end
 
+Base.eltype(r::DataFrameRow) = reduce(typejoin, eltypes(parent(r)))
```

Comment:
Should we implement this function? It should be about as costly to compute as extracting the value themselves, so I wonder whether it won't make performance worse. I guess it depends in what cases `eltype` is used by Julia.

Reply:
`eltype` is used by Julia in collect to get type information of the target container. It is a part of *iterable* interface specification. Without it vanilla `collect(dfr)` returns a vector of type `Any`.

I agree it will slow things down, but then, by defalut `DataFrameRow` is seen by Julia as having type `Any` as a container (this was my original implementation produced `Vector{Any}`).

The standard fallback method is `eltype(::Type) = Any` in array.jl in Base.

Reply:
> Without it vanilla `collect(dfr)` returns a vector of type `Any`.

Does it? I'd expect it to widen the eltype as needed using `promote_typejoin`. Maybe we need to define `Base.IteratorEltype(::DataFrameRow) = Base.EltypeUnknown()`.

Reply:
OK. The `Base.IteratorEltype(::DataFrameRow) = Base.EltypeUnknown()` is an alternative good solution. I will use it.